hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/service/DirectoryDeletingService.java

```diff
@@ -125,73 +133,96 @@ public int getPriority() {
     @Override
     public BackgroundTaskResult call() throws Exception {
       if (shouldRun()) {
+        if (LOG.isDebugEnabled()) {
+          LOG.debug("Running DirectoryDeletingService");
+        }
         runCount.incrementAndGet();
-        long count = pathLimitPerTask;
+        int dirNum = 0;
+        int subDirNum = 0;
+        int subFileNum = 0;
+        long remainNum = pathLimitPerTask;
+        List<PurgePathRequest> purgePathRequestList = new ArrayList<>();
+
+        Table.KeyValue<String, OmKeyInfo> pendingDeletedDirInfo;
         try {
+          TableIterator<String, ? extends KeyValue<String, OmKeyInfo>>
```

Comment:
This introduces a TableIterator leak, which is fixed by this PR. https://github.com/apache/ozone/pull/3922
Please help review it. @JacksonYao287

Reply:
thanks @duongkame for the fix, @Xushaohong could you also take a look?

Reply:
> thanks @duongkame for the fix, @Xushaohong could you also take a look?

@duongkame Thx for the fix!  
@JacksonYao287  That fix is merged. LTGM